docs/plugins/first_plugin.md

```diff
@@ -0,0 +1,277 @@
+
+```python
+from napari.utils.notifications import show_info
+
+def show_hello_message():
+    show_info('Hello, world!')
+```
+
+*(It doesn't look like a widget yet! We're going to use napari's widget
+autogeneration capabilities to turn this function into a widget)*
+
+
+### Add a `napari.yaml` manifest
+
+Create an empty [plugin manifest](./manifest) file at `napari_hello/napari.yaml`
+We will use this file to tell napari:
+
+1. That our plugin contributes a [**command**](./contributions.html#contributions-commands)
+   (we give the command an ID of `napari-hello.say_hi`. It must start with our plugin
+   name, and be unique).
+2. The location of the function that executes the command (the `python_name`, pointing
+   to the `show_hello_message` message in the `napari_hello` module).
+3. That our plugin contributes a [**widget**](./contributions.html#contributions-widgets),
+   and that we'd like napari to **autogenerate** the widget from the command signature
+   (so we don't need to deal with any GUI code).
+
+Add the following text to `napari.yaml`:
+
+```yaml
+name: napari-hello
+contributions:
+  commands:
+    - id: napari-hello.say_hi
+      title: Say hello, world!
+      python_name: napari_hello:show_hello_message
+  widgets:
+    - command: napari-hello.say_hi  # note, same as command.id above
+      display_name: Hello World
+      autogenerate: true
+```
+
+```{tip}
+We could have put the `show_hello_message` function anywhere, it didn't
+need to go in the top `__init__.py` file. Just make sure that the `python_name`
+of the corresponding `command` in the manifest points to the correct
+`<module_name>:<function_name>`.
+```
+
+### Update `setup.cfg`
+
+Lastly, let's make a few small changes to `setup.cfg`.
+
+1. Because we are directly using the `napari.utils.notifications` API in our,
+   `show_hello_message` function, we need to add `napari` to our package
+   **`install_requires`**. (You should add *all* of your package dependencies
+   here.  Assume nothing about your user's environment!  Not even napari.)
+
+2. We need to instruct setuptools to *include* that `napari.yaml` file
+   when it bundles our package for distribution, by adding it to the
+   **`[options.package_data]`**
+
+3. In order for napari to find our plugin when it's installed in
+   the environment, we need to add a `napari.manifest` entry to our
+   **`[options.entry_points]`** that points to the `napari.yaml`
+   file we added to the `napari_hello` module.
+
+    ```{tip}
+    Entry points are a standard Python mechanism for an installed distribution to
+    advertise components it provides to be discovered and used by other code.
+
+    See the [Entry points specification][entry_points] for details.
+    ```
+
+With the above changes, your final `setup.cfg` file should look like this:
+
+```ini
+[metadata]
+name = napari-hello
+version = 0.0.1
+classifiers =
+    Framework :: napari
+
+[options]
+packages = find:
+install_requires =
+    napari
```

Comment:
no!!! 😂 That's the same as [including pyqt5 as a dep](function).  

Now that you mention it though, I don't think the pre-commit check checks for that!

Reply:
> no!!! 😂 That's the same as including pyqt5 as a dep.

🤦‍♀️ oh yeah. Does that mean that when users follow this tutorial, they'll get the message on first run about not having a backend? If so, it might be worth us explicitly mentioning this in the tutorial - so that users don't think they've missed a step.